src/java/org/apache/cassandra/index/SecondaryIndexManager.java

```diff
@@ -1845,38 +1845,30 @@ public void handleNotification(INotification notification, Object sender)
             if (notice.memtable().isEmpty())
             {
                 IndexBuildDecider.Decision decision = IndexBuildDecider.instance.onSSTableAdded(notice);
-                build(decision, notice.added);
+                build(decision, notice.added, i -> i.shouldBuildBlocking() && !i.isSSTableAttached());
```

Comment:
is there a reason to skip SAI?

Reply:
This was added during [CASSANDRA-18670](https://issues.apache.org/jira/browse/CASSANDRA-18670). That ticket does the rebuilding in `SSTableImporter`, as it's mentioned [here](https://github.com/apache/cassandra/pull/2492#discussion_r1264245433). Not sure if we can get here some other way and want to do the rebuild.

Reply:
@jasonstack any clues on this? I don't see how we could get here if it's not through `SSTableImporter `, is Astra using this in a way that would require rebuilding here?

Reply:
we don't build SAI on writer.. it should be fine.